[AndrewNguyen4] iP

AndrewNguyen4 opened the pull request.

AndrewNguyen4 committed:
Implement a skeletal version of Areus.

Greets the user and exits with Areus.java

AndrewNguyen4 committed:
Implement the Echo feature for the skeletal version of Aerus

Repeat user input, and exit when the user types "bye".
Rename variable "hline" to "dividerLine".

AndrewNguyen4 committed:
Add a list of inputs to display when requested

Aerus can keep track of user inputs (up to 100 items).
Implement printContent() function to facilitate printing one-liners.

AndrewNguyen4 committed:
Add the ability to mark as done

User can mark or unmark tasks as done.
Add Task class to represent tasks.

AndrewNguyen4 committed:
Follow the Coding Standard

Employ minor changes to follow the Coding Standard.

AndrewNguyen4 committed:
Extract constant in Aerus.java

Refactor code to increase code quality

gu0y1 commented:
Observation: The array `Task tasks[] = new Task[100];` is defined with square brackets after the variable name.

Suggestion: It is recommended to use the format `Task[] tasks = new Task[100];` for array definitions. This aligns with the Java coding standard where the array brackets should be placed next to the type, not the variable.

ThienDuc3112 commented on src/main/java/Aerus.java:
```diff
@@ -0,0 +1,72 @@
+            String[] userInputSplit = userInput.split(" ");
+
+            // Test if the input is formatted like a mark/unmark command
+            if (userInputSplit.length == 2 && userInputSplit[1].matches("\\d+(\\.\\d+)?") &&
```
Avoid complicated expression

Can split into different part

JuvenLimHY replied:
Avoid complicated expressions

these conditions are too complicated and long. these things can be done before

ThienDuc3112 commented on src/main/java/Aerus.java:
```diff
@@ -0,0 +1,72 @@
+            if (!userInput.isEmpty()) {
+                tasks[Task.tasksCount] = new Task(userInput);
+                printContent("Added task: " + userInput);
+            }
```
Practice KISSing

Could've just use `userInput.startsWith` to check

ThienDuc3112 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected static int tasksCount = 0;
```
Structure code logically

`tasksCount` should be in the same place as Task array, may cause confusion to reader

ThienDuc3112 commented on src/main/java/Aerus.java:
```diff
@@ -0,0 +1,72 @@
+
+public class Aerus {
+
+    public static final String DIVIDER_LINE = "____________________________________________________________";
```
good use of final

ThienDuc3112 reviewed (approved):
Overall lgtm

JuvenLimHY commented on src/main/java/Aerus.java:
```diff
@@ -0,0 +1,72 @@
+        printContent("Hello! I'm Aerus!\nWhat can I do for you?");
+
+        // while block handling Aerus' main functionality
+        while (true) {
```
use switch case instead of while loop

JuvenLimHY commented on src/main/java/Aerus.java:
```diff
@@ -0,0 +1,72 @@
+            if (userInput.equals("list")) {
+                if (Task.tasksCount == 0) {
+                    printContent("You don't have any tasks!");
+                    continue;
```
instead of using continue, change to a flag to break the while loop.

JuvenLimHY commented on src/main/java/Aerus.java:
```diff
@@ -0,0 +1,72 @@
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        String userInput;
+        Task tasks[] = new Task[100];
```
Avoid magic number

JuvenLimHY reviewed (commented):
LGTM but can be better

AndrewNguyen4 committed:
Introduce UI and InputHandler classes

Working towards enforcing single-responsibility principle

AndrewNguyen4 committed:
Make great changes to the structure of the project

- Added Deadline/Event/ToDo features and their corresponding classes.
- UI and User Input are now handled by separate classes.

AndrewNguyen4 committed:
Add Exception handling

Handles exception related to user input being incorrectly formatted and marking tasks outside of recorded tasks.

AndrewNguyen4 committed:
Merge branch 'branch-Level-5'

AndrewNguyen4 committed:
Organize tasks with ArrayList

Switch to using ArrayList to keep track of tasks.
Permits better task organization and delete tasks.

okkhoy commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    public static void createNewEvent(String userInput) throws InvalidCreateEventException {
+        String[] eventInfo = userInput.split("/from|/to");
+        if (eventInfo.length == 3) {
+            Event event = new Event(eventInfo[0], eventInfo[1], eventInfo[2]);
+            Aerus.tasks[Task.tasksCount-1] = event;
+            UI.printContent("Added Event: " + event.toString());
```
some of this code can be extracted to a parser class

okkhoy commented on src/main/java/InputHandler.java:
```diff
@@ -0,0 +1,65 @@
+            String[] userInputSplit = userInput.split(" ");
+
+            // Test if the input is formatted like a mark/unmark command
+            if (isMarkCommandType(userInput)) {
+                int taskNumber = parseInt(userInputSplit[1]) - 1;
+                if (taskNumber > Task.tasksCount) {
+                    throw new InvalidMarkException();
+                } else if (userInputSplit[0].equals("mark")) {
+                    Aerus.tasks[taskNumber].isDone = true;
+                    Aerus.tasks[taskNumber].printMark();
+                    return 1;
+                } else if (userInputSplit[0].equals("unmark")) {
+                    Aerus.tasks[taskNumber].isDone = false;
+                    Aerus.tasks[taskNumber].printUnmark();
+                    return 1;
```
these can be extracted as execute methods in each command class.

okkhoy commented on src/main/java/Aerus.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Aerus {
+
+    protected static Task[] tasks = new Task[100];
```
how about expanding the task list?

okkhoy reviewed (commented):
please take a look at the comments and complete the code where necessary

AndrewNguyen4 committed:
Add DataManager class

Allows for saving data after each session.
Automatically create a data.txt file if not already exists.
Automatically loads data when a session begins.

AndrewNguyen4 committed:
Merge branch 'branch-Level-6'

AndrewNguyen4 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Aerus.java
#	src/main/java/Deadline.java
#	src/main/java/InputHandler.java
#	src/main/java/ToDo.java

AndrewNguyen4 committed:
Minor code change

AndrewNguyen4 committed:
Add Increment: A-MoreOOP

Add TaskList class for containing task.
Add help command.
Refactor code to make better use of OOP paradigm.

AndrewNguyen4 committed:
Implement Search Engine

Allows user to search for tasks by description.
TaskList class now has methods to print all tasks.
Updated help command to include search.

AndrewNguyen4 committed:
Merge pull request #1 from AndrewNguyen4/branch-Level-9

[iP Week 7] Level-9 PR

AndrewNguyen4 committed:
Add JavaDoc to classes

AndrewNguyen4 committed:
Merge branch 'master' of https://github.com/AndrewNguyen4/ip

AndrewNguyen4 committed:
Add User Guide

Minor bug fix regarding delete and mark-type commands.

AndrewNguyen4 committed:
Make minor changes to docs/README.md

AndrewNguyen4 committed:
Merge branch 'branch-A-JavaDoc'
